### PR TITLE
added hindi lang

### LINKED ISSUE #139: Make the language conversion functionality
It's time that we make this functionality live and use the language base files for the same.

**Please do this in guidance of mentors**

### PR DESCRIPTION
**Pull request by iharshit009**
Closes #139

### CODE REVIEW

File: languageBase/A_LANG_CODE.json [hunk 1]
```diff
@@ -1,44 +1,43 @@
...
     "Settings": " ",
     "Language": " ",
     "Cycle length": " ",
     "Last Period": " ",
     "Alert": " ",
-    "Please enter your Name to proceed" : " ",
+    "Please enter your Name to proceed": " ",
     "OK": " ",
     "Finish": " ",
     "RESET": " "
```
Comment (sahilit) [inline languageBase/A_LANG_CODE.json]: I think this is not needed. Remove this code.

File: languageBase/en.json [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
-    "Please select your language":"Please select your language",
-    "How can we call you?": "How can we call you?",
-    "How Long is your is cycle?": "How Long is your is cycle?",
-    "When was your last period?": "When was your last period?",
-    "Do you know your cycle length?":"Do you know your cycle length?",
+    "Ovuli": "Ovuli",
+    "Pleaseselectyourlanguage": "Please select your language",
+    "Howcanwecallyou": "How can we call you?",
+    "Pleaseenteryourname": "Please enter your name",
+    "HowLongisyouriscycle": "How Long is your is cycle?",
```
Comment (sahilit) [inline languageBase/en.json]: typo

File: languageBase/en.json [hunk 3]
```diff
@@ -1,25 +1,26 @@
 {
-    "Ovuli":"Ovuli",
-    "Please select your language":"Please select your language",
-    "How can we call you?": "How can we call you?",
-    "How Long is your is cycle?": "How Long is your is cycle?",
-    "When was your last period?": "When was your last period?",
-    "Do you know your cycle length?":"Do you know your cycle length?",
+    "Ovuli": "Ovuli",
+    "Pleaseselectyourlanguage": "Please select your language",
```
Comment (sahilit) [inline languageBase/en.json]: convert every Object keys in snake case.
for e.g: `Pleaseselectyourlanguage` will be `please_select_your_language`.

File: src/screens/ProfileScreen/AvgCycleScreen.js [hunk 4]
```diff
@@ -131,9 +131,12 @@ const AvgCycle = () => {
         <Text
           style={[{ fontFamily: 'PT-Sans', fontSize: 30, fontWeight: 'bold', alignSelf: 'center' }]}
         >
-          How long
+          {/* How long */}
+        </Text>
```
Comment (sahilit) [inline src/screens/ProfileScreen/AvgCycleScreen.js:135]: remove this as it is not displaying any text

File: src/i18n.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import i18n from 'i18n-js';
+// import * as RNLocalize from ‘react-native-localize’;
+// import memoize from ‘lodash.memoize’; // Use for caching/memoize for better performance
+import en from '../languageBase/en.json';
```
Comment (sahilit) [inline src/i18n.js]: replace `en` with `English` as you are you full name convention for all other languages

**Review by sahilit — CHANGES_REQUESTED** [review]

**Comment by sahilit** [comment]
Convert every Object keys of i18n in snake case in every files.

File: languageBase/A_LANG_CODE.json (hunk 1 above)
Comment (iharshit009) [reply languageBase/A_LANG_CODE.json]: Okay

**Comment by iharshit009** [comment]
@sahilit done

**Review by sahilit — DISMISSED** [review]

File: src/screens/ProfileScreen/AvgCycleScreen.js [hunk 6]
```diff
@@ -11,7 +11,7 @@ import {
 } from 'react-native';
 import { AntDesign } from '@expo/vector-icons';
 import SmoothPicker from 'react-native-smooth-picker';
-
+import i18n from '../../i18n';
```
Comment (narender2031) [inline src/screens/ProfileScreen/AvgCycleScreen.js]: use `@/i18n` conventions to import the module. Please check another import statement in other files.

File: src/screens/ProfileScreen/DoYouKnow.js [hunk 7]
```diff
@@ -5,6 +5,7 @@ import { StyleSheet, TouchableOpacity, Text, View, Image, Dimensions } from 'rea
 import top from '@wireframes/assets/Your_Name/top.png';
 import { AntDesign } from '@expo/vector-icons';
 import { useNavigation } from '@react-navigation/native';
+import i18n from '../../i18n';
```
Comment (narender2031) [inline src/screens/ProfileScreen/DoYouKnow.js]: similar here

File: src/screens/ProfileScreen/LanguageScreen.js [hunk 8]
```diff
@@ -15,6 +14,7 @@ import SmoothPicker from 'react-native-smooth-picker';
 import { useNavigation } from '@react-navigation/native';
 import top from '@wireframes/assets/Lang_Screen/top.png';
 import bottom from '@wireframes/assets/Lang_Screen/bottom.png';
+import i18n from '../../i18n';
```
Comment (narender2031) [inline src/screens/ProfileScreen/LanguageScreen.js]: please change the import statement here

File: src/screens/ProfileScreen/LastPeriodScreen.js [hunk 9]
```diff
@@ -16,7 +16,7 @@ import { AntDesign } from '@expo/vector-icons';
 import TopImage from '../../assets/images/Last_Period/top.png';
 import { useNavigation, useRoute } from '@react-navigation/native';
 import AskLastPeriodImage from '@/assets/images/Last_Period/AskLastPeriod.png';
-
+import i18n from '../../i18n';
```
Comment (narender2031) [inline src/screens/ProfileScreen/LastPeriodScreen.js]: Here too

File: src/screens/ProfileScreen/SecondLastPeriodScreen.js [hunk 10]
```diff
@@ -14,7 +14,7 @@ import {
 // eslint-disable-next-line import/named
 import CalendarPicker, { CALENDAR_WEEK_DAYS } from 'react-native-calendar-picker';
 import { AntDesign } from '@expo/vector-icons';
-
+import i18n from '../../i18n';
```
Comment (narender2031) [inline src/screens/ProfileScreen/SecondLastPeriodScreen.js]: here too

File: src/screens/ProfileScreen/ThirdLastPeriodScreen.js [hunk 11]
```diff
@@ -14,7 +14,7 @@ import {
 // eslint-disable-next-line import/named
 import CalendarPicker, { CALENDAR_WEEK_DAYS } from 'react-native-calendar-picker';
 import { AntDesign } from '@expo/vector-icons';
-
+import i18n from '../../i18n';
```
Comment (narender2031) [inline src/screens/ProfileScreen/ThirdLastPeriodScreen.js]: here too

File: src/screens/ProfileScreen/ThirdLastPeriodScreen.js [hunk 12]
```diff
@@ -14,7 +14,7 @@ import {
...
 import TopImage from '../../assets/images/Last_Period/top.png';
```
Comment (narender2031) [inline src/screens/ProfileScreen/ThirdLastPeriodScreen.js]: use `import TopImage from @/assets/images/Last_Period/top.png`

File: src/screens/ProfileScreen/YourNameScreen.js [hunk 13]
```diff
@@ -15,6 +15,7 @@ import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view
 import top from '@wireframes/assets/Your_Name/top.png';
 import bottom from '@wireframes/assets/Your_Name/bottom.png';
 import { useNavigation } from '@react-navigation/native';
+import i18n from '../../i18n';
```
Comment (narender2031) [inline src/screens/ProfileScreen/YourNameScreen.js]: here too

File: src/screens/ProfileScreen/index.js [hunk 14]
```diff
@@ -1,13 +1,20 @@
...
+  Image,
+  Picker,
+} from 'react-native';
 
 import profile from '@wireframes/assets/Setting_screen/profile.png';
 import { Entypo } from 'react-native-vector-icons';
 import { Feather } from 'react-native-vector-icons';
 import { MaterialIcons } from 'react-native-vector-icons';
-
+import i18n from '../../i18n';
```
Comment (narender2031) [inline src/screens/ProfileScreen/index.js]: similar here

File: src/settingStorage.js [hunk 15]
```diff
@@ -0,0 +1,26 @@
+import AsyncStorage from '@react-native-community/async-storage';
```
Comment (narender2031) [inline src/settingStorage.js]: Please add this filer under service folder

**Review by narender2031 — CHANGES_REQUESTED** [review]

**Comment by narender2031** [comment]
@iharshit009 Please also remove all unwanted comments.

File: src/screens/ProfileScreen/AvgCycleScreen.js (hunk 6 above)
Comment (iharshit009) [reply src/screens/ProfileScreen/AvgCycleScreen.js]: Okay

File: src/screens/ProfileScreen/ThirdLastPeriodScreen.js (hunk 12 above)
Comment (iharshit009) [reply src/screens/ProfileScreen/ThirdLastPeriodScreen.js]: Okay

**Comment by iharshit009** [comment]
// eslint-disable-next-line import/named
// eslint-disable-next-line import/no-unresolved
/* eslint-disable import/no-unresolved */

why these are used above many import statements?

File: src/settingStorage.js (hunk 15 above)
Comment (iharshit009) [reply src/settingStorage.js]: I have deleted this file as it was not needed

**Comment by iharshit009** [comment]
@narender2031 done

**Comment by narender2031** [comment]
// eslint-disable-next-line import/named
// eslint-disable-next-line import/no-unresolved
/* eslint-disable import/no-unresolved */

@iharshit009  Please remove these comments. 
rest everything is fine

**Comment by iharshit009** [comment]
> // eslint-disable-next-line import/named
> // eslint-disable-next-line import/no-unresolved
> /* eslint-disable import/no-unresolved */
> 
> @iharshit009 Please remove these comments.
> rest everything is fine

removing them in some files gives error

**Comment by narender2031** [comment]
Ok leave it I will check that later.

**Review by narender2031 — APPROVED** [review]

**Review by sahilit — APPROVED** [review]

**Pull request merged**